### ISSUE TITLE
Runner does not seem to work well - it eventually fails to transcode

### ISSUE THREAD

**Issue by tio-trom** [Chocobozzz/PeerTube#5849]
### Describe the current behavior

I get this error:

```
[16:49:48.200] ERROR (2921750): Cannot send job progress
    err: {
      "type": "Error",
      "message": "Expected status 204, got 404. \nThe server responded: \"Unknown runner job\".\nYou may take a closer look at the logs. To see how to do so, check out this page: https://github.com/Chocobozzz/PeerTube/blob/develop/support/doc/development/tests.md#debug-server-logs",
      "stack":
          Error: Expected status 204, got 404. 
          The server responded: "Unknown runner job".
          You may take a closer look at the logs. To see how to do so, check out this page: https://github.com/Chocobozzz/PeerTube/blob/develop/support/doc/development/tests.md#debug-server-logs
              at buildRequest (/usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:84956:14)
              at makeUploadRequest (/usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:84882:9)
              at RunnerJobsCommand.postUploadRequest (/usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:85045:12)
              at RunnerJobsCommand.update (/usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:89957:17)
              at updateTranscodingProgress (/usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:91099:28)
              at Timeout._onTimeout (/usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:91106:5)
              at listOnTimeout (node:internal/timers:573:17)
              at process.processTimers (node:internal/timers:514:7)
          ----
              at /usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:84959:19
              at /usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:53845:17
              at Test._assertFunction (/usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:53831:17)
              at Test.assert (/usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:53721:27)
              at localAssert (/usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:53689:18)
              at /usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:53693:11
              at Request2.callback (/usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:53262:7)
              at /usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:53414:22
              at IncomingMessage.<anonymous> (/usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:52664:11)
              at IncomingMessage.emit (node:events:523:35)
      "res": {
        "req": {
          "method": "POST",
          "url": "https://videos.trom.tf/api/v1/runners/jobs/80737e12-b130-4fe2-a4f9-eda97dfcc49b/update",
          "headers": {}
        },
        "header": {
          "server": "nginx",
          "date": "Sat, 24 Jun 2023 14:49:48 GMT",
          "content-type": "application/problem+json; charset=utf-8",
          "content-length": "114",
          "connection": "close",
          "x-sso-wat": "You've just been SSOed",
          "x-powered-by": "PeerTube",
          "x-frame-options": "ALLOWALL",
          "tk": "N",
          "access-control-allow-origin": "*",
          "access-control-allow-credentials": "true",
          "access-control-expose-headers": "Retry-After",
          "x-ratelimit-limit": "50",
          "x-ratelimit-remaining": "46",
          "x-ratelimit-reset": "1687618190",
          "etag": "W/\"72-vuxNSGiwa/aWfj4b6CdmHYCi/mc\"",
          "content-security-policy": "upgrade-insecure-requests",
          "x-content-type-options": "nosniff",
          "x-xss-protection": "1; mode=block",
          "x-download-options": "noopen",
          "x-permitted-cross-domain-policies": "none",
          "permissions-policy": "interest-cohort=()",
          "strict-transport-security": "max-age=63072000; includeSubDomains; preload"
        },
        "status": 404,
        "text": "{\"type\":\"about:blank\",\"title\":\"Not Found\",\"detail\":\"Unknown runner job\",\"status\":404,\"error\":\"Unknown runner job\"}"
      }
    }

```

Woks well with other servers. 

The transcoding is still happening on my computer I see, in the .cache folder. 
![Screenshot_2023-06-24_16-52-52](https://github.com/Chocobozzz/PeerTube/assets/38043315/0962f3f3-16ef-485e-a566-2077c51dbffd)

But I do not see the runner being connected with my peertube:
![2023-06-24_16-54](https://github.com/Chocobozzz/PeerTube/assets/38043315/5ba67f0c-d4b6-4579-a5b5-aa7059545de2)

**Comment by tio-trom** [comment]
From what I see now, the way it works is to download a video on the runner's machine, then convert it, and upload back. Is that true? If so the internet speeds and bandwidth are crucial.

**Comment by tio-trom** [comment]
It is the same issue with the other server we tried on, where a runner is "running". It converts well and all of that, but once it finishes, it throws the same error "[14:59:20.262] ERROR (4310): Cannot send job progress" and basically no transcoding job is ever finished.  It says "pending" in the jobs and nothing will happen.

**Comment by dhk2** [comment]
Getting a bunch of these errors as well, anyone have any luck resolving them? Or perhaps at least a way to actually view the referenced logs to hunt down the issue?

The server responded: "Unknown runner job".
You may take a closer look at the logs. To see how to do so, check out this page: https://github.com/Chocobozzz/PeerTube/blob/develop/support/doc/development/tests.md#debug-server-logs
err: {
"type": "Error",
"message": "Expected status 204, got 404. \\nThe server responded: "Unknown runner job".\\nYou may take a closer look at the logs. To see how to do so, check out this page: https://github.com/Chocobozzz/PeerTube/blob/develop/support/doc/development/tests.md#debug-server-logs",
"stack":
Error: Expected status 204, got 404.
The server responded: "Unknown runner job".

**Comment by dhk2** [comment]
Trying different things without luck so far. One thing that stands out is that it's not affecting an instancee that has nothing but short videos, it has no pending videos and no errors. the 2 sites ending up in this error state both have hours long videos as part of the normal feed. 
Could have something to do with longer jobs

**Comment by tio-trom** [comment]
Hmmm true....if I try with videos under 30MB it seems to work. Anything more it seems like the server closes the connection with the runners...
```


 Cannot process job
    err: {
      "type": "Error",
      "message": "Expected status 204, got 413. \nThe server responded: \"<html>\r\n<head><title>413 Request Entity Too Large</title><script type=\"text/javascript\" src=\"/ynh_portal.js\"></script><link type=\"text/css\" rel=\"stylesheet\" href=\"/ynh_overlay.css\"><script type=\"text/javascript\" src=\"/ynhtheme/custom_portal.js\"></script><link type=\"text/css\" rel=\"stylesheet\" href=\"/ynhtheme/custom_overlay.css\"></head>\r\n<body>\r\n<center><h1>413 Request Entity Too Large</h1></center>\r\n<hr><center>nginx</center>\r\n</body>\r\n</html>\r\n\".\nYou may take a closer look at the logs. To see how to do so, check out this page: https://github.com/Chocobozzz/PeerTube/blob/develop/support/doc/development/tests.md#debug-server-logs",
      "stack":
          Error: Expected status 204, got 413. 
          The server responded: "<html>
          <head><title>413 Request Entity Too Large</title><script type="text/javascript" src="/ynh_portal.js"></script><link type="text/css" rel="stylesheet" href="/ynh_overlay.css"><script type="text/javascript" src="/ynhtheme/custom_portal.js"></script><link type="text/css" rel="stylesheet" href="/ynhtheme/custom_overlay.css"></head>
          <body>
          <center><h1>413 Request Entity Too Large</h1></center>
          <hr><center>nginx</center>
          </body>
          </html>
          ".
          You may take a closer look at the logs. To see how to do so, check out this page: https://github.com/Chocobozzz/PeerTube/blob/develop/support/doc/development/tests.md#debug-server-logs
              at buildRequest (/usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:84956:14)
              at makeUploadRequest (/usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:84882:9)
              at RunnerJobsCommand.postUploadRequest (/usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:85045:12)
              at RunnerJobsCommand.success (/usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:89992:17)
              at processWebVideoTranscoding (/usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:91164:29)
              at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
              at async processJob (/usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:91580:5)
          ----
              at /usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:84959:19
              at /usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:53845:17
              at Test._assertFunction (/usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:53831:17)
              at Test.assert (/usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:53721:27)
              at localAssert (/usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:53689:18)
              at /usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:53693:11
              at Request2.callback (/usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:53262:7)
              at IncomingMessage.<anonymous> (/usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:53414:22)
              at IncomingMessage.emit (node:events:523:35)
              at endReadableNT (node:internal/streams/readable:1367:12)
      "res": {
        "req": {
          "method": "POST",
          "url": "https://videos.trom.tf/api/v1/runners/jobs/04ca2034-e791-48ff-86df-32c83fc7935d/success",
          "headers": {}
        },
        "header": {
          "server": "nginx",
          "date": "Sat, 24 Jun 2023 19:10:16 GMT",
          "content-type": "text/html",
          "transfer-encoding": "chunked",
          "connection": "close",
          "content-security-policy": "upgrade-insecure-requests",
          "x-content-type-options": "nosniff",
          "x-xss-protection": "1; mode=block",
          "x-download-options": "noopen",
          "x-permitted-cross-domain-policies": "none",
          "x-frame-options": "ALLOWALL",
          "permissions-policy": "interest-cohort=()",
          "strict-transport-security": "max-age=63072000; includeSubDomains; preload"
        },
        "status": 413,
        "text": "<html>\r\n<head><title>413 Request Entity Too Large</title><script type=\"text/javascript\" src=\"/ynh_portal.js\"></script><link type=\"text/css\" rel=\"stylesheet\" href=\"/ynh_overlay.css\"><script type=\"text/javascript\" src=\"/ynhtheme/custom_portal.js\"></script><link type=\"text/css\" rel=\"stylesheet\" href=\"/ynhtheme/custom_overlay.css\"></head>\r\n<body>\r\n<center><h1>413 Request Entity Too Large</h1></center>\r\n<hr><center>nginx</center>\r\n</body>\r\n</html>\r\n"
      }
    }
[21:10:16.840] INFO (65827): Checking available jobs on https://videos.trom.tf


```

**Comment by tio-trom** [comment]
So I suspect this is an Nginx issue. This is my Nginx:


```
more_set_headers "X-Frame-Options : ALLOWALL";

##
# Application
##

location @api {
  proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
  proxy_set_header Host            $host;
  proxy_set_header X-Real-IP       $remote_addr;

  client_max_body_size  100k; # default is 1M

  proxy_connect_timeout 10m;
  proxy_send_timeout    10m;
  proxy_read_timeout    10m;
  send_timeout          10m;

  proxy_pass http://127.0.0.1:9000;
}

location = /api/v1/videos/upload-resumable {
  client_max_body_size    0;
  proxy_request_buffering off;
  try_files /dev/null @api;
}

location / {
  try_files /dev/null @api;
  # Include SSOWAT user panel.
  include conf.d/yunohost_panel.conf.inc;
}

location ~ ^/api/v1/videos/(upload|([^/]+/studio/edit))$ {
  limit_except POST HEAD { deny all; }

  # This is the maximum upload size, which roughly matches the maximum size of a video file.
  # Note that temporary space is needed equal to the total size of all concurrent uploads.
  # This data gets stored in /var/lib/nginx by default, so you may want to put this directory
  # on a dedicated filesystem.
  client_max_body_size                      24G; # default is 1M
  more_set_headers "X-File-Maximum-Size : 16G always"; # inform backend of the set value in bytes before mime-encoding (x * 1.4 >= client_max_body_size)

  try_files /dev/null @api;
}

location ~ ^/api/v1/(videos|video-playlists|video-channels|users/me) {
  client_max_body_size                      6M; # default is 1M
  more_set_headers "X-File-Maximum-Size : 4M always"; # inform backend of the set value in bytes before mime-encoding (x * 1.4 >= client_max_body_size)

  try_files /dev/null @api;
}

##
# Websocket
##

location @api_websocket {
  proxy_http_version 1.1;
  proxy_set_header   X-Forwarded-For $proxy_add_x_forwarded_for;
  proxy_set_header   Host            $host;
  proxy_set_header   X-Real-IP       $remote_addr;
  proxy_set_header   Upgrade         $http_upgrade;
  proxy_set_header   Connection      "upgrade";

  proxy_pass http://127.0.0.1:9000;
}

location /socket.io {
  try_files /dev/null @api_websocket;
}

location /tracker/socket {
  # Peers send a message to the tracker every 15 minutes
  # Don't close the websocket before then
  proxy_read_timeout 15m; # default is 60s

  try_files /dev/null @api_websocket;
}

# Plugin websocket routes
location ~ ^/plugins/[^/]+(/[^/]+)?/ws/ {
  try_files /dev/null @api_websocket;
}

##
# Performance optimizations
# For extra performance please refer to https://github.com/denji/nginx-tuning
##

root /home/yunohost.app/peertube/storage;
 # Enable compression for JS/CSS/HTML, for improved client load times.
 # It might be nice to compress JSON/XML as returned by the API, but
 # leaving that out to protect against potential BREACH attack.
# gzip              on;
gzip_vary         on;
gzip_types        # text/html is always compressed by HttpGzipModule
                  text/css
                  application/javascript
                  font/truetype
                  font/opentype
                  application/vnd.ms-fontobject
                  image/svg+xml;
gzip_min_length   1000; # default is 20 bytes
gzip_buffers      16 8k;
gzip_comp_level   2; # default is 1

client_body_timeout       30s; # default is 60
client_header_timeout     10s; # default is 60
send_timeout              10s; # default is 60
keepalive_timeout         10s; # default is 75
# resolver_timeout          10s; # default is 30
reset_timedout_connection on;
proxy_ignore_client_abort on;

tcp_nopush                on; # send headers in one piece
tcp_nodelay               on; # don't buffer data sent, good for small data bursts in real time

# If you have a small /var/lib partition, it could be interesting to store temp nginx uploads in a different place
# See https://nginx.org/en/docs/http/ngx_http_core_module.html#client_body_temp_path
#client_body_temp_path /var/www/peertube/storage/nginx/;

# Bypass PeerTube for performance reasons. Optional.
# Should be consistent with client-overrides assets list in /server/controllers/client.ts
location ~ ^/client/(assets/images/(icons/icon-36x36\.png|icons/icon-48x48\.png|icons/icon-72x72\.png|icons/icon-96x96\.png|icons/icon-144x144\.png|icons/icon-192x192\.png|icons/icon-512x512\.png|logo\.svg|favicon\.png|default-playlist\.jpg|default-avatar-account\.png|default-avatar-account-48x48\.png|default-avatar-video-channel\.png|default-avatar-video-channel-48x48\.png))$ {
  more_set_headers "Cache-Control : public, max-age=31536000, immutable"; # Cache 1 year

  try_files /home/yunohost.app/peertube/storage/client-overrides/$1 /var/www/peertube/client/dist/$1 @api;
}

# Bypass PeerTube for performance reasons. Optional.
location ~ ^/client/(.*\.(js|css|png|svg|woff2|otf|ttf|woff|eot))$ {
  more_set_headers "Cache-Control : public, max-age=31536000, immutable"; # Cache 1 year

  alias /var/www/peertube/client/dist/$1;
}

# Bypass PeerTube for performance reasons. Optional.
location ~ ^/static/(thumbnails|avatars)/ {
  if ($request_method = 'OPTIONS') {
    more_set_headers "Access-Control-Allow-Origin : *";
    more_set_headers "Access-Control-Allow-Methods : GET, OPTIONS";
    more_set_headers "Access-Control-Allow-Headers : Range,DNT,X-CustomHeader,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type";
    more_set_headers "Access-Control-Max-Age : 1728000"; # Preflight request can be cached 20 days
    more_set_headers "Content-Type : text/plain charset=UTF-8";
    more_set_headers "Content-Length : 0";
    return 204;
  }

  more_set_headers "Access-Control-Allow-Origin : *";
  more_set_headers "Access-Control-Allow-Methods : GET, OPTIONS";
  more_set_headers "Access-Control-Allow-Headers : Range,DNT,X-CustomHeader,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type";
  more_set_headers "Cache-Control : public, max-age=7200"; # Cache response 2 hours

  rewrite ^/static/(.*)$ /$1 break;

  root /home/yunohost.app/peertube/storage;

  try_files $uri @api;
}

location ~ ^(/static/(webseed|streaming-playlists)/private/)|^/download {
  # We can't rate limit a try_files directive, so we need to duplicate @api

  proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
  proxy_set_header Host            $host;
  proxy_set_header X-Real-IP       $remote_addr;

  proxy_limit_rate 5M;

  proxy_pass http://127.0.0.1:9000;
}

# Bypass PeerTube for performance reasons. Optional.
location ~ ^/static/(webseed|redundancy|streaming-playlists)/ {
  limit_rate_after            5M;

  # Clients usually have 4 simultaneous webseed connections, so the real limit is 3MB/s per client
  set $peertube_limit_rate    800k;

  # Increase rate limit in HLS mode, because we don't have multiple simultaneous connections
  if ($request_uri ~ -fragmented.mp4$) {
    set $peertube_limit_rate  5M;
  }

  # Use this line with nginx >= 1.17.0
  limit_rate $peertube_limit_rate;
  # Or this line if your nginx < 1.17.0
  # set $limit_rate $peertube_limit_rate;

  if ($request_method = 'OPTIONS') {
    more_set_headers "Access-Control-Allow-Origin : *";
    more_set_headers "Access-Control-Allow-Methods : GET, OPTIONS";
    more_set_headers "Access-Control-Allow-Headers : Range,DNT,X-CustomHeader,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type";
    more_set_headers "Access-Control-Max-Age : 1728000"; # Preflight request can be cached 20 days
    more_set_headers "Content-Type : text/plain charset=UTF-8";
    more_set_headers "Content-Length : 0";
    return 204;
  }

  if ($request_method = 'GET') {
    more_set_headers "Access-Control-Allow-Origin : *";
    more_set_headers "Access-Control-Allow-Methods : GET, OPTIONS";
    more_set_headers "Access-Control-Allow-Headers : Range,DNT,X-CustomHeader,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type";

    # Don't spam access log file with byte range requests
    access_log off;
  }

  # Enabling the sendfile directive eliminates the step of copying the data into the buffer
  # and enables direct copying data from one file descriptor to another.
  sendfile on;
  sendfile_max_chunk 1M; # prevent one fast connection from entirely occupying the worker process. should be > 800k.
  aio threads;

  rewrite ^/static/webseed/(.*)$ /videos/$1 break;
  rewrite ^/static/(.*)$         /$1        break;

  root /home/yunohost.app/peertube/storage;

  try_files $uri @api;
}

```

I suspect we need to increase some values there....

**Comment by dhk2** [comment]
Hmm, you seem to be missing this section from the current example config,  I pushed the max up to 999G to see if the errors stopped , testing now.
 `
  location ~ ^/api/v1/runners/jobs/[^/]+/(update|success)$ {
    client_max_body_size                      12G; # default is 1M
    add_header            X-File-Maximum-Size 8G always; # inform backend of the set value in bytes before mime-encoding (x * 1.4 >= client_max_body_size)

    try_files /dev/null @api;
  }`

https://github.com/Chocobozzz/PeerTube/blob/develop/support/nginx/peertube

**Comment by tio-trom** [comment]
> Hmm, you seem to be missing this section from the current example config, I pushed the max up to 999G to see if the errors stopped , testing now.
> 
> > ```
> >  location ~ ^/api/v1/runners/jobs/[^/]+/(update|success)$ {
> >       client_max_body_size                      12G; # default is 1M
> >       add_header            X-File-Maximum-Size 8G always; # inform backend of the set value in bytes before mime-encoding (x * 1.4 >= client_max_body_size)
> >   
> >       try_files /dev/null @api;
> >     }
> > ```
> 
> https://github.com/Chocobozzz/PeerTube/blob/develop/support/nginx/peertube

Ah ok. Let me know how it goes for you. I'll try to test it later on today as well.

**Comment by dhk2** [comment]
Upping all the maximums in nginx didn't help, but this seems to have fixed it. I was missing this new section of production.yaml

```
  > remote_runners:
  >   # Consider jobs that are processed by a remote runner as stalled after this period of time without any update
  >   stalled_jobs:
  >     live: '30 seconds'
  >     vod: '2 minutes'
```

By changing stalled_jobs vod to 10 minutes and restarting peertube it stopped going back to pending in Runner Jobs admin and started showing progress. Watching on the runner machine it was taking was about 4-5 minutes for the "Built supported ffmpeg encoders." line to show up after getting the job.

This forum thread pointed the way https://framacolibri.org/t/unknown-runner-job-on-remote-runner/18013/2

**Comment by tio-trom** [comment]
I did all of those changes. It seems to work a bit better. Maybe? But it fails even for like 7 min 300MB videos.

```
[20:23:11.681] INFO (494771): Checking available jobs on https://videos.trom.tf
[20:23:11.967] ERROR (494771): Cannot abort job after error
    err: {
      "type": "Error",
      "message": "Expected status 204, got 404. \nThe server responded: \"Unknown runner job\".\nYou may take a closer look at the logs. To see how to do so, check out this page: https://github.com/Chocobozzz/PeerTube/blob/develop/support/doc/development/tests.md#debug-server-logs",
      "stack":
          Error: Expected status 204, got 404. 
          The server responded: "Unknown runner job".
          You may take a closer look at the logs. To see how to do so, check out this page: https://github.com/Chocobozzz/PeerTube/blob/develop/support/doc/development/tests.md#debug-server-logs
              at buildRequest (/usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:84956:14)
              at makePostBodyRequest (/usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:84898:10)
              at RunnerJobsCommand.postBodyRequest (/usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:85038:12)
              at RunnerJobsCommand.error (/usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:89971:17)
              at /usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:91777:25
          ----
              at /usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:84959:19
              at /usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:53845:17
              at Test._assertFunction (/usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:53831:17)
              at Test.assert (/usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:53721:27)
              at localAssert (/usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:53689:18)
              at /usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:53693:11
              at Request2.callback (/usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:53262:7)
              at /usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:53414:22
              at IncomingMessage.<anonymous> (/usr/lib/node_modules/@peertube/peertube-runner/dist/peertube-runner.js:52664:11)
              at IncomingMessage.emit (node:events:523:35)
      "res": {
        "req": {
          "method": "POST",
          "url": "https://videos.trom.tf/api/v1/runners/jobs/470c32a7-913a-4ade-890d-4029e3a962a9/error",
          "data": {
            "message": "Expected status 204, got 404. \nThe server responded: \"Unknown runner job\".\nYou may take a closer look at the logs. To see how to do so, check out this page: https://github.com/Chocobozzz/PeerTube/blob/develop/support/doc/development/tests.md#debug-server-logs",
            "jobToken": "ptrjt-4b890077-a652-4e0a-a9cd-b3d8d9572586",
            "runnerToken": "ptrt-ad482e3e-130e-46bc-8614-eca6aa21d28b"
          },
          "headers": {
            "content-type": "application/json"
          }
        },
        "header": {
          "server": "nginx",
          "date": "Sun, 25 Jun 2023 18:23:11 GMT",
          "content-type": "application/problem+json; charset=utf-8",
          "content-length": "114",
          "connection": "close",
          "x-sso-wat": "You've just been SSOed",
          "x-powered-by": "PeerTube",
          "x-frame-options": "ALLOWALL",
          "tk": "N",
          "access-control-allow-origin": "*",
          "access-control-allow-credentials": "true",
          "access-control-expose-headers": "Retry-After",
          "etag": "W/\"72-vuxNSGiwa/aWfj4b6CdmHYCi/mc\"",
          "content-security-policy": "upgrade-insecure-requests",
          "x-content-type-options": "nosniff",
          "x-xss-protection": "1; mode=block",
          "x-download-options": "noopen",
          "x-permitted-cross-domain-policies": "none",
          "permissions-policy": "interest-cohort=()",
          "strict-transport-security": "max-age=63072000; includeSubDomains; preload"
        },
        "status": 404,
        "text": "{\"type\":\"about:blank\",\"title\":\"Not Found\",\"detail\":\"Unknown runner job\",\"status\":404,\"error\":\"Unknown runner job\"}"
      }
    }

```

**Comment by tio-trom** [comment]
Wow it works now. I tested with a 1.3GB file. 1h long video. Converted in less than 20 minutes via my local computer. This is amazing! My internet connection even dropped at one point but that didn't seem to stop the process. So all of the above changes that we made to our nginx and production.yml files are still needed?

**Comment by Chocobozzz** [comment]
I think we try to still upgrade the job after we sent a success. I'll try to improve this in a future runner version, but it should hurt. Closing this issue as root cause should be fixed by 0.0.5 runner version.

**Comment by tio-trom** [comment]
Last update about the last video that had some errors, after restarting the runners they picked up the job and finished it. Overall it is working but it still needs some improvement!
